src/services/download/mod.rs

```diff
@@ -202,7 +202,7 @@ fn download_future_stream(
 ///
 /// This is common functionality used by all many downloaders.
 async fn download_stream(
-    source: SourceFileId,
+    source: &str,
```

Comment:
Are we passing this in purely for the trace log?  Do we even have trace log enabled?

But if we're keeping this, why can't we pass `&SourceFileId`?